client/src/hooks/fetchJobs.jsx

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+import React, { useReducer, useEffect } from 'react';
+const ACTIONS = {
+  MAKE_REQUEST: 'make_request',
+  GET_DATA: 'get-data',
+  ERROR: 'error',
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case ACTIONS.MAKE_REQUEST:
+      return { loading: true, jobs: [] };
+    case ACTIONS.GET_DATA:
+      return { ...state, loading: false, jobs: action.payload.jobs };
+    case ACTIONS.ERROR:
+      return {
+        ...state,
+        loading: false,
+        error: action.payload.error,
+        jobs: [],
+      };
+    default:
+      return state;
+  }
+};
+
+const fetchJobs = (params, page) => {
+  const [state, dispatch] = useReducer(reducer, {
+    jobs: [],
+    loading: true,
+    error: false,
+  });
+
+  useEffect(() => {
+    const cancel_Token = axios.CancelToken.source();
```

Comment:
What's the purpose of the cancel_Token?

Reply:
it's for real time search, so while someone's typing it cancels the search calls between letters . This way I just type out the search term and it does the search instead of me having to click "search" which does the request.